cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/efs/AwsEfsFileSystem.java

```diff
@@ -173,4 +183,30 @@ public String getThroughputMode() {
     public void setThroughputMode(String throughputMode) {
         this.throughputMode = throughputMode;
     }
+
+    public List<String> getAssociatedInstanceGroupNames() {
+        return associatedInstanceGroupNames;
+    }
+
+    public void setAssociatedInstanceGroupNames(List<String> associatedInstanceGroupNames) {
+        this.associatedInstanceGroupNames = associatedInstanceGroupNames;
+    }
+
+    private void setDefaultIfNeeded() {
+        if (backupPolicyStatus == null) {
+            backupPolicyStatus = BACKUP_POLICY_STATUS_DEFAULT;
+        }
+
+        if (performanceMode == null) {
+            performanceMode = PERFORMANCE_MODE_DEFAULT;
+        }
+
+        if (throughputMode == null) {
+            throughputMode = THROUGHPUT_MODE_DEFAULT;
+        }
```

Comment:
What is the benefit for doing this instead of the former default fallback init in the constructor?

Reply:
check style does not allow that. The function is too complicated. NPath Complexity exceeds the limit. I have to break it into another function to lower the complexity of the constructor function